sys/arvo.hoon

```diff
@@ -502,101 +515,232 @@
+            who=(unit ship)
+            eny=(unit @)
+            bod=(unit vase)
+        ==
+    ::  larval Arvo structural interface
+    ::
+    |%
+    ++  come  ^come                                     ::  22
+    ++  load  ^load                                     ::  46
+    ::
+    ::  XX my preference would be for +peek to no-op here
+    ::  XX falling thru because %vega scrys for hoon-version
+    ::  XX revisit
+    ::
+    ++  peek  ^peek                                     ::  47
+    ::
+    ++  poke  |=  *                                     ::  10
+              ^-  [(list ovum) *]
+              =>  .(+< ((hard ,[now=@da ovo=ovum]) +<))
+              ^-  [(list ovum) *]
+              =.  +>.$
+                ?+  -.q.ovo
+                  ::  ignore unrecognized
+                  ::
+                  ~&  [%larval-ignore p.ovo -.q.ovo]
+                  +>.$
+                ::  install %zuse or vane
+                ::
+                    %veer
+                  ^+  +>.$
+                  ::  use the maximum comet if we don't know who we are yet
+                  ::
+                  =/  our  ?^(who u.who (dec (bex 128)))
+                  =.  ..veer  (veer our now q.ovo)
+                  +>.$(bod ?^(bod bod `bud.^poke))
+                ::  add entropy
+                ::
+                    %wack
+                  ^+  +>.$
+                  ?>  ?=(@ q.q.ovo)
+                  +>.$(eny `q.q.ovo)
+                ::  become who you were born to be
+                ::
+                    %whom
+                  ^+  +>.$
+                  ?>  ?=(@ q.q.ovo)
+                  +>.$(who `q.q.ovo)
+                ==
+              ::  upgrade once we've accumulated identity, entropy, and %zuse
+              ::
+              ?.  &(?=(^ who) ?=(^ eny) ?=(^ bod))
+                [~ +>.$]
+              ~>  %slog.[0 leaf+"arvo: metamorphosis"]
+              =/  nyf
+                (turn vanes.^poke |=([label=@tas =vane] [label vase.vane]))
+              (load u.who now u.eny ova=~ u.bod nyf)
+    ::
+    ++  wish  |=  txt=*                                 ::  4
+              ?>  ?=(@ txt)
+              q:(slap ?~(bod pit u.bod) (ream txt))
+    --
+::
+::  persistent arvo state
+::
+=/  pit=vase  !>(..is)                                  ::
+=/  vil=vile  (viol p.pit)                              ::  cached reflexives
+=|  $:  lac=?                                           ::  laconic bit
+        eny=@                                           ::  entropy
+        our=ship                                        ::  identity
+        bud=vase                                        ::  %zuse
         vanes=(list [label=@tas =vane])                 ::  modules
     ==                                                  ::
-=<  |%
-    ++  come  |=  {@ (list ovum) pone}                  ::  22
+=<  ::  Arvo structural interface
+    ::
+    |%
+    ++  come  |=  {@ @ @ (list ovum) vise pone}         ::  22
```

Comment:
Can we put some faces on these atoms? Same for +load

Reply:
Oh, maybe it's fine since it just wraps `^come`.

Reply:
Right. It's worth noting that `+come` and `+load` are the only typed methods of the arvo core -- they're only intended to be called on a new Arvo by the old Arvo during an upgrade. I've re-used that mechanism for the larval-stage upgrade.